.github/workflows/aio-preview-build.yml

```
@@ -0,0 +1,42 @@
+# This workflow builds the AIO previews for pull requests when a certain label is applied.
+# The actual deployment happens as part of a dedicated second workflow to avoid security
+# issues where the building would otherwise occur in an authorized context where secrets
+# could be leaked. More details can be found here:
+
+# https://securitylab.github.com/research/github-actions-preventing-pwn-requests/.
+
+name: Build AIO app for preview deployment
+
+on:
+  pull_request:
+    types: [synchronize, labeled]
```

```suggestion
    types: [opened, reopened, synchronize, labeled]
```

Should we also do it on opened so that when someone on the team creates a pull request with the label already applied it runs?

---

I remember testing this back when I did it for COMP. Confirmed `labeled` emits when a PR is opened with initial labels: https://github.com/angular/components/pull/26167.